src/app/components/plugin-manager-settings.js

```diff
@@ -60,75 +60,83 @@ export class PluginManagerSettings {
   }
 
   /** Clear one permission from a plugin */
-  clearPersmission (from, to) {
-    if (!this.permissions[from]) return
-    delete this.permissions[from][to]
-    if (Object.keys(this.permissions[from]).length === 0) {
-      delete this.permissions[from]
+  clearPersmission (from, to, method) {
+    if (this.permissions[to] && this.permissions[to][method]) {
+      delete this.permissions[to][method][from]
+      if (Object.keys(this.permissions[to][method]).length === 0) {
+        delete this.permissions[to][method]
+      }
+      if (Object.keys(this.permissions[to]).length === 0) {
+        delete this.permissions[to]
+      }
+      yo.update(this.currentSetting, this.settings())
     }
-    yo.update(this.currentSetting, this.settings())
   }
 
   /** Clear all persmissions from a plugin */
-  clearAllPersmission (from) {
-    if (!this.permissions[from]) return
-    delete this.permissions[from]
+  clearAllPersmission (to) {
+    if (!this.permissions[to]) return
+    delete this.permissions[to]
     yo.update(this.currentSetting, this.settings())
   }
 
   settings () {
-    const permissionByModule = (key, permission) => {
-      const permissionByPlugin = (name, plugin) => {
-        function updatePermission () {
-          plugin.allow = !plugin.allow
+    const permissionByToPlugin = (toPlugin, funcObj) => {
+      const permissionByFunctions = (funcName, fromPlugins) => {
+        function updatePermission (plugin) {
+          self.permissions[toPlugin][funcName][plugin].allow = !self.permissions[toPlugin][funcName][plugin].allow
         }
-        const checkbox = plugin.allow
-          ? yo`<input onchange="${updatePermission}" type="checkbox" checked id="permission-${name}" aria-describedby="module ${key} ask permission for ${name}" />`
-          : yo`<input onchange="${updatePermission}" type="checkbox" id="permission-${name}" aria-describedby="module ${key} ask permission for ${name}" />`
-
-        return yo`
-        <div class="form-group ${css.permissionKey}">
-          <div class="${css.checkbox}">
-            ${checkbox}
-            <label for="permission-${name}" data-id="pluginManagerSettingsPermission${name}">Allow <u>${name}</u> to access <u>${key}</u></label>
-          </div>
-          <i onclick="${() => this.clearPersmission(key, name)}" class="fa fa-trash-alt" data-id="pluginManagerSettingsRemovePermission${name}"></i>
-        </div>`
+        let self = this
```

Comment:
You should need "self" if you use arrow functions

Reply:
I know. But im second level ot wasn't accessible. Don't know why :(

Reply:
As I commented above, I think it's because you use the keyword `function` instead of creating an arrow function `() => `